display_list/dl_canvas.h

```diff
@@ -62,7 +62,8 @@ class DlCanvas {
   virtual void Save() = 0;
   virtual void SaveLayer(std::optional<const DlRect>& bounds,
                          const DlPaint* paint = nullptr,
-                         const DlImageFilter* backdrop = nullptr) = 0;
+                         const DlImageFilter* backdrop = nullptr,
+                         std::optional<int64_t> backdrop_id = std::nullopt) = 0;
```

Comment:
The backdrop filter argument is a DlImageFilter, which is a mirror of the dart:ui ImageFilter object. It doesn't make sense for the backdrop id to be a property of the image filter as 1) a single filter object can be used in multiple places in the same scene and 2) can be used in contexts for which the backdropId propety doesn't make sense like image filters.

I think instead it would make more sense to refactor saveLayer to group both the backdrop image filter and the backdrop id into one BackdropData struct.